src/utils/filterUtils/filterByCategory.js

```diff
@@ -0,0 +1,8 @@
+export const filterByCategory = (products, category) => {
+  if (Object.values(category).every((v) => !v)) {
+    return products;
+  }
+  return products.filter((product) => {
+    return category[product.catagoryName];
+  });
+};
```

Comment:
suggestion: add comments so that it's helpful, to understand what this function in doing

Reply:
Thank you for your time, I will rectify the code.